### ISSUE TITLE
Support region_mask for multi-resolution maps

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1715]
This is a reminder issue to make region_mask work for multi-resolution maps:

https://github.com/gammapy/gammapy/blob/5759d4f2fefda7792340c970d0f87e3d2e25dc59/gammapy/maps/wcs.py#L815

Probably only a few lines of code extra or changed are needed for this, but it needs an expert in gammapy.maps that knows how multi-resolution maps and map coords work.

**Comment by AtreyeeS** [comment]
This is true not just for `region_mask`, but for many other functions as well.
For example, `Map.cutout()` definitely does not work correctly for multi-resolution maps.

**Comment by AtreyeeS** [comment]
Support for multi-resolution maps will be done in future. Closing now
